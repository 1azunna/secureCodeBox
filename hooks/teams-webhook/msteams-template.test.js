const { getMessageCardByTemplate } = require("./msteams-template");

const vulnerabilityManagement = {
  enabled: true,
  name: "Kibana Dashboard",
  dashboardUrl: "htps://dashboard.yourservice/",
  dashboardFindingsUrl: "https://dashboard.yourservice/filter:09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
};

test("Test 1", async () => {

  const scan = {
    metadata: {
      uid: "09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
      name: "demo-scan-1601086432",
      namespace: "my-scans",
      labels: {
        company: "iteratec",
        "attack-surface": "external",
      },
    },
    spec: {
      scanType: "Nmap",
      parameters: ["-Pn", "localhost"],
    },
    status: {
      findingDownloadLink:
        "https://my-secureCodeBox-instance.com/scan-b9as-sdweref--sadf-asdfsdf-dasdgf-asdffdsfa7/findings.json",
      findings: {
        categories: {
          "A Client Error response code was returned by the server": 1,
          "Information Disclosure - Sensitive Information in URL": 1,
          "Strict-Transport-Security Header Not Set": 1,
        },
        count: 3,
        severities: {
          high: 10,
          medium: 5,
          low: 2,
          informational: 1,
        },
      },
      finishedAt: "2020-05-25T02:38:13Z",
      rawResultDownloadLink:
        "https://my-secureCodeBox-instance.com/scan-blkfsdg-sdgfsfgd-sfg-sdfg-dfsg-gfs98-e8af2172caa7/zap-results.json?Expires=1601691232",
      rawResultFile: "zap-results.json",
      rawResultType: "zap-json",
      state: "Done",
    },
  };

  expect(getMessageCardByTemplate(scan, vulnerabilityManagement) ).toMatchSnapshot();
});

test("Test 2", async () => {

  const scan = {
    metadata: {
      uid: "09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
      name: "demo-scan-1601086432",
      namespace: "my-scans",
      labels: {
        company: "iteratec",
        "attack-surface": "external",
      },
    },
    spec: {
      scanType: "Nmap",
      parameters: ["-Pn", "localhost"],
    },
    status: {
      findingDownloadLink:
        "https://my-secureCodeBox-instance.com/scan-b9as-sdweref--sadf-asdfsdf-dasdgf-asdffdsfa7/findings.json",
      findings: {
        categories: {},
        count: 0,
        severities: {},
      },
      finishedAt: "2020-05-25T02:38:13Z",
      rawResultDownloadLink:
        "https://my-secureCodeBox-instance.com/scan-blkfsdg-sdgfsfgd-sfg-sdfg-dfsg-gfs98-e8af2172caa7/zap-results.json?Expires=1601691232",
      rawResultFile: "zap-results.json",
      rawResultType: "zap-json",
      state: "Done",
    },
  };

  expect(getMessageCardByTemplate(scan, vulnerabilityManagement) ).toMatchSnapshot();
});

test("Test 3", async () => {

  const scan = {
    metadata: {
      uid: "09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
      name: "demo-scan-1601086432",
      namespace: "my-scans",
      labels: {
        company: "iteratec",
        "attack-surface": "external",
      },
    },
    spec: {
      scanType: "Nmap",
      parameters: ["-Pn", "localhost"],
    },
    status: {
      findingDownloadLink:
        "https://my-secureCodeBox-instance.com/scan-b9as-sdweref--sadf-asdfsdf-dasdgf-asdffdsfa7/findings.json",
      findings: {},
      finishedAt: "2020-05-25T02:38:13Z",
      rawResultDownloadLink:
        "https://my-secureCodeBox-instance.com/scan-blkfsdg-sdgfsfgd-sfg-sdfg-dfsg-gfs98-e8af2172caa7/zap-results.json?Expires=1601691232",
      rawResultFile: "zap-results.json",
      rawResultType: "zap-json",
      state: "Done",
    },
  };

  expect(getMessageCardByTemplate(scan, vulnerabilityManagement) ).toMatchSnapshot();
});
