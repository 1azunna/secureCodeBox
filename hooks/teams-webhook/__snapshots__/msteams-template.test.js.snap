// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Kibana Dashboard URL contains the parsed scan UID 1`] = `
Object {
  "@context": "https://schema.org/extensions",
  "@type": "MessageCard",
  "sections": Array [
    Object {
      "activityImage": "https://docs.securecodebox.io/img/Favicon.svg",
      "activitySubtitle": "Created at 2021-01-01T14:29:25Z",
      "activityTitle": "Scan: **'demo-scan-1601086432'**",
      "facts": Array [],
      "startGroup": true,
      "text": "__Findings Severity Overview:__",
    },
    Object {
      "facts": Array [],
      "text": "__Findings Category Overview:__",
    },
  ],
  "summary": "Scan 09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
  "themeColor": "0078D7",
  "title": "New Nmap security scan results are available!",
}
`;

exports[`Test if a  scan (with empty categories / severities object) will be rendered properly 1`] = `
Object {
  "@context": "https://schema.org/extensions",
  "@type": "MessageCard",
  "sections": Array [
    Object {
      "activityImage": "https://docs.securecodebox.io/img/Favicon.svg",
      "activitySubtitle": "Created at 2021-01-01T14:29:25Z",
      "activityTitle": "Scan: **'demo-scan-1601086432'**",
      "facts": Array [],
      "startGroup": true,
      "text": "__Findings Severity Overview:__",
    },
    Object {
      "facts": Array [],
      "text": "__Findings Category Overview:__",
    },
  ],
  "summary": "Scan 09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
  "themeColor": "0078D7",
  "title": "New Nmap security scan results are available!",
}
`;

exports[`Test if a  scan (without categories / severities objects) will be rendered properly 1`] = `
Object {
  "@context": "https://schema.org/extensions",
  "@type": "MessageCard",
  "sections": Array [
    Object {
      "activityImage": "https://docs.securecodebox.io/img/Favicon.svg",
      "activitySubtitle": "Created at 2021-01-01T14:29:25Z",
      "activityTitle": "Scan: **'demo-scan-1601086432'**",
      "facts": Array [],
      "startGroup": true,
      "text": "__Findings Severity Overview:__",
    },
    Object {
      "facts": Array [],
      "text": "__Findings Category Overview:__",
    },
  ],
  "summary": "Scan 09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
  "themeColor": "0078D7",
  "title": "New Nmap security scan results are available!",
}
`;

exports[`Test if a complete scan (with findings / categories / severities) will be rendered properly 1`] = `
Object {
  "@context": "https://schema.org/extensions",
  "@type": "MessageCard",
  "sections": Array [
    Object {
      "activityImage": "https://docs.securecodebox.io/img/Favicon.svg",
      "activitySubtitle": "Created at 2021-01-01T14:29:25Z",
      "activityTitle": "Scan: **'demo-scan-1601086432'**",
      "facts": Array [
        Object {
          "name": "high",
          "value": 10,
        },
        Object {
          "name": "medium",
          "value": 5,
        },
        Object {
          "name": "low",
          "value": 2,
        },
        Object {
          "name": "informational",
          "value": 1,
        },
      ],
      "startGroup": true,
      "text": "__Findings Severity Overview:__",
    },
    Object {
      "facts": Array [
        Object {
          "name": "A Client Error response code was returned by the server",
          "value": 1,
        },
        Object {
          "name": "Information Disclosure - Sensitive Information in URL",
          "value": 1,
        },
        Object {
          "name": "Strict-Transport-Security Header Not Set",
          "value": 1,
        },
      ],
      "text": "__Findings Category Overview:__",
    },
  ],
  "summary": "Scan 09988cdf-1fc7-4f85-95ee-1b1d65dbc7cc",
  "themeColor": "0078D7",
  "title": "New Nmap security scan results are available!",
}
`;
